nbodykit/dask/test_tpm.py

```diff
@@ -0,0 +1,75 @@
+from nbodykit.extensionpoints import DataSource
+from binary_file import BinaryFile
+from fileiter import FileIterator
+from dask.delayed import delayed
+from dask import compute
+
+import numpy
+
+class DaskTPMSnapshot(DataSource):
+    """
+    DataSource to read snapshot files from Martin White's TPM simulations
+    """
+    plugin_name = "TPMSnapshot"
+    columns = ['Position', 'Velocity']
+    
+    def __init__(self, path, BoxSize, rsd=None, bunchsize=4*1024*1024):
+        
+        self.path = path
+        self.BoxSize = BoxSize
+        self.rsd = rsd
+        self.bunchsize = bunchsize
+        
+        dtypes = [('Position', ('f4', 3)), ('Velocity', ('f4', 3)), ('ID', 'u8')]
+        self.storage = BinaryFile(self.path, dtypes, header_size=28)
```

Comment:
For example, GADGET uses F77 unformatted blocks, then you need a 'gap_size' parameter at least. 

Talking about GADGET it is far more complicated. It may be possible to peek the header to tell the schema, e.g. if the Position is in f8, if the ID is in u4. In Gadget these are stored as attributes in the header file.

It gets even more complicated when we allow multiple particle types. The column length is variable in this case! For example,  position contains all data types, but gas temperature only contains the gas particles. Gadget also has a stupid optimization that he mass of dark matter particles is sometimes not stored in the freaking files, but as a single number in the header block, while all other particles are stored in the mass column!